local/code/datums/actions/mob/sing_tones.dm

```diff
@@ -0,0 +1,48 @@
+/datum/action/sing_tones
+	name = "Sing Tones"
+	desc = "Use your internal synthesizer to sing!"
+	button_icon = 'icons/obj/art/musician.dmi'
+	button_icon_state = "xylophone"
+	var/datum/song/song
+	/// What instruments can be used.
+	var/allowed_instrument_ids = list("spaceman", "meowsynth", "square", "sine", "saw")
+	/// Instruments added after being emagged.
+	var/emag_instrument_ids = list("bikehorn", "honk")
+	/// Set to TRUE if already emagged.
+	var/emagged = FALSE
+
+/datum/action/sing_tones/Grant(mob/grant_to)
+	..()
+	RegisterSignal(grant_to, COMSIG_SPECIES_LOSS, PROC_REF(on_species_loss))
+	RegisterSignal(grant_to, COMSIG_ATOM_EMAG_ACT, PROC_REF(on_emag_act))
+	song = new(grant_to, allowed_instrument_ids, 15)
+	if(isethereal(grant_to))
+		desc = "Use your electric discharger to sing!"
+
+/datum/action/sing_tones/Remove(mob/remove_from)
+	..()
```

Comment:
call parent at the end of `Remove` instead of the beginning and change to `. = ..()`